tests/issues/test_issue_383.py

```diff
@@ -0,0 +1,25 @@
+import heterocl as hcl
+
+def test_complex_select():
+
+    hcl.init(hcl.Int(32))
+    a = hcl.placeholder((10,), name="a")
+    b = hcl.placeholder((10,), name="b")
+    c = hcl.placeholder((10,), name="c")
+    d = hcl.placeholder((10,), name="d")
+    
+    def kernel_select(a, b, c, d):
+        use_imm = hcl.scalar(1)
+        with hcl.for_(0, 10, name="i") as i:
+            src = hcl.select(use_imm == 1, hcl.cast(hcl.Int(16), (c[i] + b[i])), 
+                                           hcl.cast(hcl.Int(32), (c[i] - b[i]))
+                                           )
+            dst = hcl.cast(hcl.Int(32), (2 * (c[i] + b[i])))
+            d[i] = hcl.select(dst >= (-1 * src),
+                    hcl.select(dst <= src, a[i], src),
+                    (-1 * src))
+    s = hcl.create_schedule([a, b, c, d], kernel_select)
+    hcl.build(s, target="vhls")
```

Comment:
Can you also test the generated code (using `assert`)?

Reply:
Right, I've added that. The complex `hcl.select` turns into a long expression, so I added an assert to count there are right number of `?` characters in the code string

Reply:
There are three `hcl.select` here, but why do we have 6 `?` ?

Reply:
Great question, I also stopped and thought about that: 

Eventually we build a large assignment stmt out of this line (because `src` and `dst` are inlined):
```python
d[i] = hcl.select(dst >= (-1 * src), hcl.select(dst <= src, a[i], src),(-1 * src))
```
here `src` itself is a `hcl.select` value, so whenever it appears it has one `?`, and there are four. Plus the two `?` from `hcl.select`, we have 6 in total